provisioner/shell/provisioner.go

```diff
@@ -232,7 +231,9 @@ func (p *Provisioner) Provision(ctx context.Context, ui packer.Ui, comm packer.C
 		}
 
 		p.config.envVarFile = tf.Name()
-		defer os.Remove(p.config.envVarFile)
+
+		// Don't think this is needed as there is already a call to delete tf.Name() after creating the file
+		//defer os.Remove(p.config.envVarFile)
```

Comment:
Seeing as `p.config.envVarFile` is pointing to the local file I don't think we need this as its deletion is being deferred on line 221. But let me know if I am missing something.

Reply:
This seems good but I'd rather you just delete the line instead of commenting it out.